packages/blade/src/components/Charts/LineChart/LineChart.web.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import {
+  LineChart as RechartsLineChart,
+  Line as RechartsLine,
+  ResponsiveContainer as RechartsResponsiveContainer,
+} from 'recharts';
+import { useChartsColorTheme } from '../utils';
+import type { ChartLineProps, ChartLineWrapperProps } from './types';
+import { useTheme } from '~components/BladeProvider';
+import { metaAttribute } from '~utils/metaAttribute';
+import BaseBox from '~components/Box/BaseBox';
+import getIn from '~utils/lodashButBetter/get';
+import type { DataAnalyticsAttribute, TestID } from '~utils/types';
+import { makeAnalyticsAttribute } from '~utils/makeAnalyticsAttribute';
+
+const ChartLine: React.FC<ChartLineProps> = ({
+  color,
+  strokeStyle = 'solid',
+  type = 'monotone',
+  dot = false,
+  activeDot = true,
+  showLegend = true,
+  _index,
+  _colorTheme,
+  ...props
+}) => {
+  const { theme } = useTheme();
+  const themeColors = useChartsColorTheme({ colorTheme: _colorTheme ?? 'default' });
+  const colorToken = color ? getIn(theme.colors, color) : themeColors[_index ?? 0];
+
+  const strokeDasharray =
+    strokeStyle === 'dashed' ? '5 5' : strokeStyle === 'dotted' ? '2 2' : undefined;
+
+  const isLineDotted = strokeStyle === 'dashed';
+  const animationBegin = isLineDotted
+    ? theme.motion.delay.gentle + theme.motion.duration.xgentle
+    : theme.motion.delay.gentle;
+  const animationDuration = theme.motion.duration.xgentle;
+
+  return (
+    <RechartsLine
+      stroke={colorToken}
+      strokeWidth={3}
+      strokeDasharray={strokeDasharray}
+      type={type}
+      activeDot={activeDot}
+      dot={dot}
+      legendType={showLegend ? 'line' : 'none'}
+      animationBegin={animationBegin}
+      animationDuration={animationDuration}
+      {...props}
+    />
+  );
+};
+
+// Main components
+const ChartLineWrapper: React.FC<ChartLineWrapperProps & TestID & DataAnalyticsAttribute> = ({
+  children,
+  colorTheme = 'default',
+  testID,
+  data,
+  ...restProps
+}) => {
+  const lineChartModifiedChildrens = React.useMemo(() => {
+    let LineChartIndex = 0;
+    return React.Children.map(children, (child) => {
+      if (React.isValidElement(child) && child.type === ChartLine) {
```

Comment:
We don't use child.type because it's brittle and can fail in certain situations post bundling right? 

Can we use componentId checks?